backend/amp-live-list.go

```diff
@@ -157,13 +165,71 @@ func createMetadata(r *http.Request) []BlogPosting {
 	return result
 }
 
-func createLiveBlogSample(newStatus int, timestamp time.Time, r *http.Request) LiveBlogSample {
+func createLiveBlogSample(newStatus int, timestamp time.Time, r *http.Request, page Page) LiveBlogSample {
 	if newStatus > len(blogs) {
 		newStatus = len(blogs)
 	}
 	blogItems := getBlogEntries(newStatus, timestamp)
 	score := createScore(newStatus, 0)
-	return LiveBlogSample{BlogItems: blogItems, FootballScore: score, BlogMetadata: createMetadata(r)}
+	firstBlogID := strings.TrimPrefix(r.URL.Query().Get("from"), BLOG_ID_PREFIX)
+	firstItemIndex := getBlogEntryIndexFromID(firstBlogID, blogItems)
+	lenghtCurrentPageBlog := int(math.Min(float64(len(blogItems)), float64(firstItemIndex+MAX_BLOG_ITEMS_NUMBER_PER_PAGE)))
+
+	return LiveBlogSample{BlogItems: blogItems[firstItemIndex:lenghtCurrentPageBlog],
+		FootballScore: score,
+		BlogMetadata:  createMetadata(r),
+		NextPageURL:   getNextPageUrl(firstItemIndex, r, page, blogItems),
+		PrevPageURL:   getPrevPageUrl(firstItemIndex, r, page, newStatus),
+		PageNumber:    getPageNumberFromProductIndex(firstItemIndex)}
+}
+
+func getNextPageUrl(firstItemIndex int, r *http.Request, page Page, blogItems []BlogItem) string {
+	nextPageFirstItemIndex := firstItemIndex + MAX_BLOG_ITEMS_NUMBER_PER_PAGE
+	nextPageId := getNextPageId(r, page, blogItems, nextPageFirstItemIndex)
+	if nextPageId != "" {
+		return buildPaginationURL(r, page, nextPageId)
```

Comment:
You build the URL twice instead of:

```
urlPrefix := buildURL(...)
nextIndex := nextId(...)
previousIndex := prevId(...)
nextUrl := urlPrefix + nextIndex
prevUrl := urlPrefix + prevIndex
```